modulemd/v2/include/modulemd-2.0/modulemd-module-stream.h

```diff
@@ -82,7 +85,7 @@ struct _ModulemdModuleStreamClass
 
   /* Padding to allow adding up to 8 new virtual functions without
    * breaking ABI. */
-  gpointer padding[8];
+  gpointer padding[10];
```

Comment:
This number must decrease when adding a new function. The padding exists to ensure that the data structure size remains the same when adding new functions.

Reply:
Oh! I didn't decrement this in any of the new functions I added :/. I'll fix it if you want me to. I thought that array helps to accommodate new virtual functions.

So how will we add any new virtual functions when the padding will reach 0?

Reply:
I just fixed that in ModulemdComponent and ModulemdDefaults. If you know of any other places besides that, they should also be fixed.

If we run out of padding, we need to make a new major release (libmodulemd 3.x). We will need to bump the soname to libmodulemd.so.3 and any package that needs to consume libmodulemd will need to be rebuilt against the new library. This is something to avoid as much as possible (ideally only when it's discovered that the current API has a critical flaw that can only be solved by replacing it entirely, such as what happened between 1.x and 2.x). Hence, we keep the padding in place to reduce the risk of this. It's unlikely that we will hit zero on any of them, though.

Reply:
So its not needed anywhere else since the others weren't getting inherited.